### ISSUE TITLE
Property use function will run even when property is not set

### ISSUE THREAD

**Issue by vmrose** [eivindfjeldstad/validate#29]
Consider the following validation:

``` js
user
  .path('salutation')
  .type('string', 'Salutation has the wrong type')
  .required(false)
  .use(function(value) {
    return validator.isIn(value, ['M', 'F']) // Using the validator module
  }, 'Salutation can only be either M or F')
user
  .path('name')
  .type('string')
  .required(true)
user
  .path('surname')
  .type('string')
  .required(true)
```

Note that the field `salutation` is not required.

If you validate this object:

``` js
{
  name: 'Vincent'
  surname: 'M'
}
```

You'll get an error with the message defined by the `.use` function.

Should I implement every custom use function with the possibility that the value might be `undefined`?
If so, it would be nice to clarify this with a short sentence in the readme.

Thanks for your awesome validation module, I really like it.

**Comment by eivindfjeldstad** [comment]
Thank you! Yeah, that's the way it currently works. I can't think of a case where validating an undefined value would be useful, so I'm open to changing this behavior in the next major release. I'll add a note to the readme in the meantime:)

**Comment by eivindfjeldstad** [comment]
fixed in 4.0.0
